src/v/cloud_roles/apply_credentials.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "cloud_roles/types.h"
+#include "http/client.h"
+
+#include <memory>
+
+namespace cloud_roles {
+
+class apply_credentials {
+public:
+    class impl {
```

Comment:
can you add this in a commit, and then gcp and aws apply in separate commits?

Reply:
moved the commit structure around a bit so these are introduced separately